### PR TITLE
Created NewsModel,  NewsFeedViewModel and NetworkManager

### PR DESCRIPTION
**Pull request by Meendex**
Researched JSON data
- Path to the data

Created NewsModel structs
Added structs:
- struct Welcome
- struct WelcomeData
- struct Child
- struct NewsModel
Added CodingKeys for:
- commentsCount = "num_comments"
- thumbnailHeight = "thumbnail_height"
- thumbnailWidth = "thumbnail_width"

Created NetworkManager class
Function:
- func getNewsAsync()

Created NewsFeedViewModel class
Protocols:
- protocol NewsViewModelIn
- protocol NewsViewModelOut
Created init()
function in extension:
- func getNewsAsync()

### CODE REVIEW

**Review by fenix56 — APPROVED** [review]
Looks good, well done.

File: HomePageFeedApp/NewsFeed/NewsModel.swift [hunk 1]
```diff
@@ -2,11 +2,38 @@
 //  NewsModel.swift
 //  HomePageFeedApp
 //
-//  Created by Przemek on 17/09/2022.
+//  Created by Przemek & Mindaugas B on 17/09/2022.
 //
 
 import Foundation
 
+struct Welcome: Decodable {
```
Comment (codemanager255) [inline HomePageFeedApp/NewsFeed/NewsModel.swift:10]: Give meaning full model name

File: HomePageFeedApp/NewsFeed/NewsModel.swift [hunk 2]
```diff
@@ -2,11 +2,38 @@
...
+struct WelcomeData: Decodable {
+    let after: String
+    let children: [Child]
+}
+
+struct Child: Decodable {
+    let data: NewsModel
+}
+
 struct NewsModel: Decodable {
```
Comment (codemanager255) [inline HomePageFeedApp/NewsFeed/NewsModel.swift:23]: No need to put model as suffix

File: HomePageFeedApp/NewsFeed/NewsFeedViewModel.swift [hunk 3]
```diff
@@ -2,16 +2,39 @@
...
+extension NewsFeedViewModel: NewsViewModelIn {
+
+    func getNewsAsync() async {
+        do {
+            let welcome = try await
+            networkManager.getNewsAsync()
+            newsList = welcome.data.children
+            
+        } catch {
+            print(error)
```
Comment (codemanager255) [inline HomePageFeedApp/NewsFeed/NewsFeedViewModel.swift:37]: handle error in UI , show alert message.

**Pull request merged**